astroplan/core.py

```diff
@@ -162,56 +195,359 @@ def altaz(self, time, target=None, obswl=None):
+        else:
+            raise ValueError('`rise_set` keyword may only be "rising" or '
+                             '"setting"')
+
+        if sum(condition) < 1:
+            raise ValueError('Target does not rise/set with respect to '
+                             '`horizon` within 24 hours')
+
+        nearest_index = np.argwhere(condition)[0][0]
+
+        # Address special cases when nearest point is end point in array:
+        if nearest_index == 0:
+            lower_limit = t[0]
+            upper_limit = t[1]
+            lower_alt = alt[0]
+            upper_alt = alt[1]
+        elif nearest_index == len(alt):
+            lower_limit = t[-2]
+            upper_limit = t[-1]
+            lower_alt = alt[-2]
+            upper_alt = alt[-1]
+        # Normal cases:
+        elif alt[nearest_index] > horizon:
+            lower_limit = t[nearest_index-1]
+            upper_limit = t[nearest_index]
+            lower_alt = alt[nearest_index-1]
+            upper_alt = alt[nearest_index]
+        else:
+            lower_limit = t[nearest_index]
+            upper_limit = t[nearest_index+1]
+            lower_alt = alt[nearest_index]
+            upper_alt = alt[nearest_index+1]
+
+        return (lower_limit, upper_limit), (lower_alt, upper_alt)
+
+    @u.quantity_input(horizon=u.deg)
+    def _two_point_interp(self, times, altitudes, horizon=0*u.deg):
+        '''
+        Do linear interpolation between two ``altitudes`` at
+        two ``times`` to determine the time where the altitude
+        goes through zero.
+
+        Parameters
+        ----------
+        times : `~astropy.time.Time`
+            Two times for linear interpolation between
+
+        altitudes : array of `~astropy.units.Quantity`
+            Two altitudes for linear interpolation between
+
+        horizon : `~astropy.units.Quantity`
+            Solve for the time when the altitude is equal to
+            reference_alt.
+
+        Returns
+        -------
+        t : `~astropy.time.Time`
+            Time when target crosses the horizon
+
+        '''
+        slope = (altitudes[1] - altitudes[0])/(times[1].jd - times[0].jd)
+        return Time(times[1].jd - ((altitudes[1] - horizon)/slope).value,
+                    format='jd')
+
+    def _altitude_trig(self, LST, target):
+        '''
+        Calculate the altitude of ``target`` at local sidereal times ``LST``.
+
+        This method provides a factor of ~3 speed up over calling `altaz`, and
+        inherently does *not* take the atmosphere into account.
+
+        Parameters
+        ----------
+        LST : `~astropy.time.Time`
+            Local sidereal times (array)
+
+        target : {`~astropy.coordinates.SkyCoord`, `FixedTarget`} or similar
+            Target celestial object's coordinates.
+
+        Returns
+        -------
+        alt : `~astropy.unit.Quantity`
+            Array of altitudes
+        '''
+        alt = np.arcsin(np.sin(self.location.latitude.radian) *
```

Comment:
I did some quick tests and I do see it as a fixed-time overhead (as you thought it should be, @cdeil) - not sure why it's different.  Presumably the extra cost over `radian` comes from having to do the extra bits to create a `Quantity`.